docs/public/specifications/tests/DNSSEC-TP/dnssec07.md

```diff
@@ -1,40 +1,326 @@
+   ("DNSKEY Query").
+
+3. Create a [DNSSEC Query] with query type DS and query name *Child Zone*
+   ("DS Query") (for the parent name servers).
+
+4. Retrieve all name server names and IP addresses for *Child Zone* using methods
+   [Get-Del-NS-Names-and-IPs] and [Get-Zone-NS-Names-and-IPs] ("Child NS IP").
+
+5.  Create the following empty sets:
+
+    1.  Name server IP address ("Ignored Child NS").
+    2.  Name server IP address ("No Response DNSKEY Query").
+    3.  Name server IP address ("No Auth DNSKEY Response").
+    4.  Name server IP address and [RCODE Name] ("Error RCODE DNSKEY Response").
+    5.  Name server IP address ("No DNSKEY").
+    6.  Name server IP address ("Signed Response").
+    7.  Name server IP address ("Ignored Parent NS IP").
+    8.  Name server IP address ("No DS").
+    9.  Name server IP address ("DS in Response").
+
+6.  For each name server IP address in *Child NS IP* do:
+
+    1.  Send *SOA Query* to the name server IP.
+    2.  Add the name server IP to the *Ignored Child NS* set and go to next name
+        server IP if at least one of the following criteria is met:
+          1. There is no DNS response.
+          2. The [RCODE Name] in the response is not "NoError".
+          3. The AA flag is not set in the response.
+          4. The SOA record is missing in the answer section.
+    3.  Send *DNSKEY Query* to the name server IP.
+    4.  Add the name server IP to the *No Response DNSKEY Query* set and go to
+        next name server IP if there is no DNS response to the query.
+    5.  Add the name server IP to the *No Auth DNSKEY Response* set and go to
+        next name server IP if the AA bit is not set in the response.
+    6.  Add the name server IP and the [RCODE Name] to the
+        *Error RCODE DNSKEY Response* set and go to next name server IP if the
+        [RCODE Name] is not NOERROR.
+    7.  Add the name server IP to the *Signed Response* set and go to next name
+        server IP if the answer section contains the following DNS records:
+        * At least one DNSKEY record.
+        * At least one RRSIG record covering the DNSKEY RRset (verification not
+          done in this test case).
+    8.  Else add the name server IP to the *No DNSKEY* set and go to
+        next name server IP.
+
+7.  Retrieve all name server IP addresses for the parent of *Child Zone* using
+    method [Get-Parent-NS-IP] ("Parent NS IP").
+
+8.  If *Undelegated DS* is non-empty then do:
+
+    1. Add name server IP as "-" to the *DS in Response* set.
+    2. Make *Parent NS IP* an empty set.
+
+9.  If the *Signed Response* set is empty, make *Parent NS IP* and
+    *DS in Response* empty sets.
+
+>   Note: *Parent NS IP* will be empty if any of the followng is true:
+>     * *Undelegated test* is TRUE.
+>     * *Undelegated DS* is non-empty.
+>     * *Child Zone* is ".", i.e. root zone.
+>     * The *Signed Response* set is empty.
```

Comment:
The bullet point list does not render properly on GitHub:

<img width="997" height="65" alt="image" src="https://github.com/user-attachments/assets/59136d46-5a6b-4d8b-af7a-adb243d94483" />